packages/mikado_reborn/src/mixins/Button/Button.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  CreateElement, VNode, VNodeChildren, VNodeData,
+} from 'vue';
+import { Component, Vue, Prop } from 'vue-property-decorator';
+import { MkrIcon } from '../../components/Icon';
+import './Button.scss';
+
+@Component({
+  components: {
+    MkrIcon,
+  },
+})
+export default class Button extends Vue {
+  @Prop({ default: false })
+  disabled!: boolean;
+
+  @Prop({
+    default: 'medium',
+    validator: (size) => ['small', 'medium', 'large'].includes(size),
+  })
+  size!: 'small' | 'medium' | 'large';
+
+  @Prop()
+  icon?: string;
+
+  @Prop({
+    default: 'left',
+    validator: (side) => ['left', 'right'].includes(side),
+  })
+  iconSide!: 'left' | 'right';
+
+  class = 'mkr__button';
+
+  // eslint-disable-next-line class-methods-use-this
+  get classes(): VNodeData['class'] {
+    return {};
+  }
+
+  private get hasContent(): boolean {
+    return Boolean(this.$slots.default?.length);
+  }
+
+  click(event: Event): void {
+    this.$emit('click', event);
+  }
+
+  render(createElement: CreateElement): VNode {
+    const staticClasses = [this.class];
+    const content: VNodeChildren = [this.$slots.default];
+
+    staticClasses.push(`${this.class}--${this.size}`);
+
+    if (!this.hasContent) staticClasses.push(`${this.class}--icon-only`);
+
+    if (this.icon) {
+      const icon = createElement(MkrIcon, {
+        props: { name: this.icon },
+        staticClass: `${this.class}__icon--${this.iconSide}`,
+      });
+      content[this.iconSide === 'left' ? 'unshift' : 'push'](icon);
```

Comment:
I think this one is easier to read (probably because i did not know it could be written the way you did) : 
```suggestion
      content = this.iconSide === 'left' ? [icon, content] : [content, icon]
```

Reply:
Sounds good to me, i'll do the following as content is an array :
```suggestion
      content = this.iconSide === 'left' ? [icon, ...content] : [...content, icon]
```